pyanaconda/ui/tui/spokes/__init__.py

```diff
@@ -184,17 +184,21 @@ def prompt(self, entry = None):
             return _("Enter new value for '%s' and press enter\n") % entry.title
 
     def input(self, entry, key):
-        if entry.aux.match(key):
+        valid, err_msg = check_name(key)
+        if entry.aux.match(key) and valid:
```

Comment:
These two conditions should be swapped as checking the ``valid`` boolean is cheaper than trying to do a regexp match.